geo/geomfn: implement ST_AsBinary({geometry,text})

hueypark opened the pull request:
Fixes #48379, #48870

Release note (sql change): Implemented the geometry based builtins `ST_AsBinary` with encoding.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49481)
<!-- Reviewable:end -->

hueypark commented:
@otan I think we also need 'st_ashexwkb' which receives encoding as an additional variable.

otan commented:
Yep, there is an issue (#48880) for that.

This LGTM!

bors r+

otan commented:
looks like ./pkg/geo doesn't compile under test.

otan commented on pkg/sql/sem/builtins/geo_builtins.go:
```diff
@@ -560,13 +560,51 @@ var geoBuiltins = map[string]builtinDefinition{
+			},
+			Info: infoBuilder{
+				info: "Returns the WKB representation of a given Geography. " +
+					"The second variant takes a second argument denoting the encoding - using little-endian ('NDR') or big-endian ('XDR') encoding.",
```
can you make this but just "This variant has a second argument denoting the encoding"

hueypark replied:
Done.

hueypark committed:
geo/geomfn: implement ST_AsBinary({geometry,text})

Fixes #48379, #48870

Release note (sql change): Implemented the geometry based builtins
`ST_AsBinary` with encoding.

hueypark commented:
> Yep, there is an issue (#48880) for that.

#48880 is for `st_ashexewkb`. We need `st_ashexwkb`(without e).

> looks like ./pkg/geo doesn't compile under test.

It's corrected.

otan commented:
Ah. Yeah that's reasonable - PostGIS doesn't seem to have it but I don't
thonk it hurts to include it.

On Sat, 23 May 2020, 4:38 pm Jaewan Park, <notifications@github.com> wrote:

> Yep, there is an issue (#48880
> <https://github.com/cockroachdb/cockroach/issues/48880>) for that.
>
> #48880 <https://github.com/cockroachdb/cockroach/issues/48880> is for
> st_ashexewkb. We need st_ashexwkb(without e).
>
> looks like ./pkg/geo doesn't compile under test.
>
> It's corrected.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/49481#issuecomment-633154372>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA32FQ52GWUT67QYJ5BTQ4DRTBM7BANCNFSM4NIMZ5XA>
> .
>

hueypark commented:
@otan Would file an issue for `st_ashexwkb` and merge this? I will fix that later.

otan commented:
bors r+

otan commented:
bors r+

otan commented:
bors r+

hueypark commented:
@otan I looked through PostGIS document and we should not make `st_ashexwkb` with encoding.
If you make it, it breaks compatibility.

FYI https://postgis.net/docs/RT_ST_AsHexWKB.html
